library/core/src/future/poll_once.rs

```diff
@@ -0,0 +1,32 @@
+use crate::fmt;
+use crate::future::Future;
+use crate::pin::Pin;
+use crate::task::{Context, Poll};
+
+/// Resolves to the output of polling a future once.
+///
+/// This `struct` is created by [`poll_once()`]. See its
+/// documentation for more.
+#[unstable(feature = "future_poll_once", issue = "92115")]
+pub struct PollOnce<F> {
+    pub(crate) future: F,
```

Comment:
Idea to support `!Unpin`:

```suggestion
    pub(crate) future: Option<F>,
```